### ISSUE TITLE
Exceptions while trying to benchmark node

### ISSUE THREAD

**Issue by quickchase** [hyperledger/besu#4779]
Running the latest binary release, I attempted to do some quick benchmarking of a Besu full node and I noticed that the node was throwing some exceptions.

I downloaded [ethspam](https://github.com/shazow/ethspam) and [versus](https://github.com/INFURA/versus) and tried to run it against the node:

```bash
$ ./ethspam --rpc http://localhost:8545 | ./versus --stop-after 10000 --concurrency=10 http://localhost:8545
Endpoints:

0. "http://localhost:8545"

   Requests:   1001.00 per second, 0.99 per second for errors
   Timing:     0.0100s avg, 0.0002s min, 1.3084s max
               0.0733s standard deviation

   Percentiles:
     25% in 0.0007s
     50% in 0.0013s
     75% in 0.0026s
     90% in 0.0054s
     95% in 0.0146s
     99% in 0.1354s

   Errors: 0.03%
     3 × "bad status code: 400"

** Summary for 1 endpoints:
   Completed:  10000 results with 10000 total requests
   Timing:     9.990057ms request avg, 10.305638047s total run time
   Errors:     3 (0.03%)
   Mismatched: 0
```

You can see that there were some errors, that's unexpected running against localhost...

I then grabbed some logs from the node:

```
2022-12-05 18:05:16.164+00:00 | vert.x-worker-thread-12 | ERROR | MainnetTransactionProcessor | Critical Exception Processing Transaction
org.hyperledger.besu.ethereum.trie.MerkleTrieException: Unable to load trie node value for hash 0xc4252200a1075c1e91304cdcdc9d941dd92253287f0de434c7919354e75cb261 location 0x
        at org.hyperledger.besu.ethereum.trie.StoredNode.lambda$load$0(StoredNode.java:133)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at org.hyperledger.besu.ethereum.trie.StoredNode.load(StoredNode.java:131)
        at org.hyperledger.besu.ethereum.trie.StoredNode.accept(StoredNode.java:63)
        at org.hyperledger.besu.ethereum.trie.StoredMerklePatriciaTrie.get(StoredMerklePatriciaTrie.java:119)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateKeyValueStorage.getStorageValueBySlotHash(BonsaiWorldStateKeyValueStorage.java:200)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiPersistedWorldState.getStorageValueBySlotHash(BonsaiPersistedWorldState.java:421)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiLayeredWorldState.getStorageValueBySlotHash(BonsaiLayeredWorldState.java:156)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.getStorageValueBySlotHash(BonsaiWorldStateUpdater.java:353)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.getStorageValue(BonsaiWorldStateUpdater.java:340)
        at org.hyperledger.besu.ethereum.bonsai.BonsaiAccount.getStorageValue(BonsaiAccount.java:208)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
        at org.hyperledger.besu.evm.operation.SLoadOperation.execute(SLoadOperation.java:58)
        at org.hyperledger.besu.evm.EVM.runToHalt(EVM.java:282)
        at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.codeExecute(AbstractMessageProcessor.java:161)
        at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.process(AbstractMessageProcessor.java:173)
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.process(MainnetTransactionProcessor.java:492)
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:404)
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:169)
        at org.hyperledger.besu.ethereum.transaction.TransactionSimulator.processWithWorldUpdater(TransactionSimulator.java:227)
        at org.hyperledger.besu.ethereum.transaction.TransactionSimulator.process(TransactionSimulator.java:157)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthCall.resultByBlockHash(EthCall.java:72)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.AbstractBlockParameterOrBlockHashMethod.handleParamTypes(AbstractBlockParameterOrBlockHashMethod.java:85)
        at org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.EthCall.response(EthCall.java:96)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.BaseJsonRpcProcessor.process(BaseJsonRpcProcessor.java:42)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.TracedJsonRpcProcessor.process(TracedJsonRpcProcessor.java:41)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.TimedJsonRpcProcessor.process(TimedJsonRpcProcessor.java:45)
        at org.hyperledger.besu.ethereum.api.jsonrpc.execution.JsonRpcExecutor.execute(JsonRpcExecutor.java:92)
        at org.hyperledger.besu.ethereum.api.handlers.JsonRpcExecutorHandler.lambda$handler$8(JsonRpcExecutorHandler.java:79)
        at io.vertx.ext.web.impl.BlockingHandlerDecorator.lambda$handle$0(BlockingHandlerDecorator.java:48)
        at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
        at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

Node Config:
```
ExecStart=/home/eth/besu/bin/besu \
    --network=MAINNET \
    --data-path=/home/eth/besu/data \
    --data-storage-format=BONSAI \
    --Xbonsai-use-snapshots=false \
    --sync-mode=X_SNAP \
    --pruning-enabled=true \
    --rpc-http-enabled=true \
    --rpc-http-apis=ADMIN,DEBUG,ETH,NET,TRACE,TXPOOL,WEB3 \
    --rpc-http-host=0.0.0.0 \
    --rpc-http-port=8545 \
    --rpc-http-cors-origins="*" \
    --host-allowlist="*" \
    --rpc-ws-enabled=true \
    --rpc-ws-apis=ADMIN,DEBUG,ETH,NET,TRACE,TXPOOL,WEB3 \
    --rpc-ws-host=0.0.0.0 \
    --rpc-ws-port=8546 \
    --metrics-enabled=true \
    --metrics-host=0.0.0.0 \
    --metrics-port=6060 \
    --engine-rpc-enabled=true \
    --engine-rpc-port=8551 \
    --engine-jwt-secret=/home/eth/jwtsecret \
    --engine-host-allowlist="*" \
    --Xplugin-rocksdb-high-spec-enabled=true
```

**Comment by non-fungible-nelson** [comment]
Handling this under issue #4785

**Comment by non-fungible-nelson** [comment]
Closing for bonsai refactor fixes in #5123, please re-open if you experience this on latest.
